homeassistant/components/epson/media_player.py

```diff
@@ -82,24 +85,21 @@ class EpsonProjectorMediaPlayer(MediaPlayerEntity):
     )
 
     def __init__(
-        self, projector: Projector, name: str, unique_id: str | None, entry: ConfigEntry
+        self, projector: Projector, unique_id: str, entry: ConfigEntry
     ) -> None:
         """Initialize entity to control Epson projector."""
         self._projector = projector
         self._entry = entry
-        self._attr_name = name
         self._attr_available = False
         self._cmode = None
         self._attr_source_list = list(DEFAULT_SOURCES.values())
         self._attr_unique_id = unique_id
-        if unique_id:
```

Comment:
Because this is set in the config entry. And if it wasn't present, there's a function here that restores that unique id if it wasn't set for some reason

Reply:
Right, but looking upstream, it seems that the method that gets the serial can return `None`?

Reply:
fyi projector returns serial only if it's turned on and only if it is newer model. Olders models don't return anything on serial request.

Reply:
So older models don't have any unique_id?

Reply:
Ok, sorry, my review comment earlier https://github.com/home-assistant/core/pull/98164#pullrequestreview-1575659064 might have led down an unhelpful path.  If there is a scenario that can result in SN=`None`, then the check is needed, but how would all the other parameters (model mfr etc be set in that case)?

Maybe set `manufacturer` and `model` in all cases, and set `identifiers`, `via_device` only when you have a unique id during init?  

Sorry, my understanding of how this all works is not complete.

Reply:
I haven't seen a device without identifiers yet, so I am not 100% sure if that works like expected. My current suggestion is to use the entry id if there is no unique id available, what do we think about that?

Reply:
Indeed if the projector is 'busy' [it could return SN=`None`](https://github.com/pszafer/epson_projector/blob/f9efe665e3a3b379bf1d7f637b87b66ac18a7ddd/epson_projector/projector_serial.py#L141) 

Can a pytest be added to verify that everything still works if no serial number is returned?  It feels like this is a scenario we need to remain compliant with in the future.